Content.Server/Gatherable/GatherableSystem.cs

```diff
@@ -49,51 +45,49 @@ private void OnInteractUsing(EntityUid uid, GatherableComponent component, Inter
             BreakOnTargetMove = true,
             BreakOnUserMove = true,
             MovementThreshold = 0.25f,
+            DuplicateCondition = DuplicateConditions.SameTarget,
         };
 
         _doAfterSystem.TryStartDoAfter(doAfter);
     }
 
-    private void OnDoAfter(EntityUid uid, GatherableComponent component, GatherableDoAfterEvent args)
+    private void OnInteractHand(EntityUid uid, GatherableComponent component, InteractHandEvent args)
     {
-        if(!TryComp<GatheringToolComponent>(args.Args.Used, out var tool))
+        if (component.ToolWhitelist?.Tags?.Contains("Hand") == false)
             return;
 
-        tool.GatheringEntities.Remove(uid);
+        var doAfter = new DoAfterArgs(args.User, TimeSpan.FromSeconds(component.HarvestTimeByHand), new GatherableDoAfterEvent(), uid, target: uid)
+        {
+            BreakOnDamage = true,
+            BreakOnTargetMove = true,
+            BreakOnUserMove = true,
+            MovementThreshold = 0.25f,
+            DuplicateCondition = DuplicateConditions.SameTarget,
+        };
+
+        _doAfterSystem.TryStartDoAfter(doAfter);
+    }
+
+    private void OnDoAfter(EntityUid uid, GatherableComponent component, GatherableDoAfterEvent args)
+    {
+        if (TryComp<GatheringToolComponent>(args.Args.Used, out var tool))
+            tool.GatheringEntities.Remove(uid);
+
         if (args.Handled || args.Cancelled)
             return;
 
-        Gather(uid, args.Args.Used, component, tool.GatheringSound);
+        Gather(uid, args.Args.Used, component, tool?.GatheringSound);
         args.Handled = true;
     }
 
-    public void Gather(EntityUid gatheredUid, EntityUid? gatherer = null, GatherableComponent? component = null, SoundSpecifier? sound = null)
+    public void Gather(EntityUid gatheredUid, EntityUid? gatherer, GatherableComponent? component = null, SoundSpecifier? sound = null)
     {
         if (!Resolve(gatheredUid, ref component))
             return;
 
         // Complete the gathering process
-        _destructible.DestroyEntity(gatheredUid);
-        _audio.PlayPvs(sound, gatheredUid);
-
-        // Spawn the loot!
-        if (component.MappedLoot == null)
-            return;
-
-        var pos = Transform(gatheredUid).MapPosition;
-
-        foreach (var (tag, table) in component.MappedLoot)
-        {
-            if (tag != "All")
-            {
-                if (gatherer != null && !_tagSystem.HasTag(gatherer.Value, tag))
-                    continue;
-            }
-            var getLoot = _prototypeManager.Index<EntityLootTablePrototype>(table);
-            var spawnLoot = getLoot.GetSpawns();
-            var spawnPos = pos.Offset(_random.NextVector2(0.3f));
-            Spawn(spawnLoot[0], spawnPos);
-        }
+        _destructible.DestroyEntity(gatheredUid, gatherer);
+        _audio?.PlayPvs(sound, gatheredUid);
```

Comment:
Just call some method on miningsystem instead for now rather than adding the entityuid to destroyentity.

Reply:
I don't understand how to do this, please advise
for now i've gone back to the naive implementation with the user and tool in destroy args

Reply:
I don't believe I have the necessary experience to resolve this, so I'm going to remove the destructible functionality and have spaceshrooms simply be an item that spawns on the ground